tests/processors/test_bioscan_pool_xp_to_traction_processor.py

```diff
@@ -0,0 +1,126 @@
+from unittest.mock import MagicMock, patch
+import pytest
+
+from lab_share_lib.processing.base_processor import BaseProcessor
+from tol_lab_share.processors.bioscan_pool_xp_to_traction_processor import BioscanPoolXpToTractionProcessor
+
+
+@pytest.fixture
+def subject(config):
+    return BioscanPoolXpToTractionProcessor(config)
+
+
+@pytest.fixture(autouse=True)
+def traction_reception_message():
+    with patch("tol_lab_share.processors.bioscan_pool_xp_to_traction_processor.TractionReceptionMessage") as trm:
+        yield trm.return_value
+
+
+class TestBioscanPoolXpToTractionProcessor:
+    """Tests for the BioscanPoolXpToTractionProcessor class.
+    Note, however, that the tests of the process method are more like integration tests, checking that the entire
+    process works. Those tests are therefore not mocking out the dependencies apart from the TractionReceptionMessage so
+    that we can check the correct message would go to Traction.
+    """
+
+    def test_is_subclass_of_base_processor(self):
+        assert issubclass(BioscanPoolXpToTractionProcessor, BaseProcessor)
+
+    def test_can_be_initialised(self, config):
+        assert BioscanPoolXpToTractionProcessor(config) is not None
+
+    def test_instantiate_returns_instance(self, config):
+        instance = BioscanPoolXpToTractionProcessor.instantiate(MagicMock(), MagicMock(), config)
+        assert isinstance(instance, BioscanPoolXpToTractionProcessor)
+
+    def test_instantiate_passes_arguments_to_init(self, config):
+        with patch(
+            "tol_lab_share.processors.bioscan_pool_xp_to_traction_processor.BioscanPoolXpToTractionProcessor.__init__"
+        ) as init_mock:
+            init_mock.return_value = None
+            BioscanPoolXpToTractionProcessor.instantiate(MagicMock(), MagicMock(), config)
+
+        init_mock.assert_called_with(config)
+
+    def test_process_returns_true_for_valid_message(self, subject, valid_bioscan_pool_xp_to_traction_message):
+        assert subject.process(valid_bioscan_pool_xp_to_traction_message) is True
+
+    def test_process_populated_traction_message_correctly(
```

Comment:
Just putting this out there: I've seen people using an assertion library called [`assertpy`](https://github.com/assertpy/assertpy). It does the same thing underneath but it adds upto some nice syntax .

Reply:
This is quite a nice looking library.  I wasn't aware of it.  I'm not sure we should introduce it now given that we have 350 ish tests that should be updated to use it.  Well worth considering for any new Python projects though where the effort and risk would be lower.  Then adopt it in projects like this if it is a success.

Reply:
Yeah, I've seen people using it. I like its syntax.